### ISSUE TITLE
Paragraph breaks are lost in subsections

### ISSUE THREAD

**Issue by gustavdelius** [r-lib/pkgdown#991]
The following roxygen block illustrates the issue. This leads to a two-paragraph subsection in the Rd file, but pkgdown removes the paragraph break when converting this to html, where both are put on the same line as "First paragraph Second paragraph".
```
#' Testing subsections
#' 
#' @section A section:
#' \subsection{A subsection}{
#'   First paragraph
#'   
#'   Second paragraph
#' }
#' @name testing
```

**Comment by jayhesselberth** [comment]
Possibly related to #819.

**Comment by gustavdelius** [comment]
It is indeed related to #819. In fact, any paragraph break within braces is removed by pkgdown. For example
```
#' Hello
#'
#' testing
#'
#' \strong{One paragraph
#'
#' another paragraph}
```
Gives two bold paragraphs in the .Rd file but only one bold line in the .html file produced by pkgdown.

**Comment by kevinushey** [comment]
Perhaps related, but line breaks are also lost within pre-formatted code chunks, e.g.

```
#' \preformatted{
#' line 1
#'
#' line 2
#'
#' line 3
#' }
```

Instead, each newline is indented:

<img width="138" alt="Screen Shot 2019-04-09 at 4 35 52 PM" src="https://user-images.githubusercontent.com/1976582/55841802-99736d00-5ae5-11e9-8ba1-4431f6f8b1de.png">

**Comment by hadley** [comment]
Would someone mind creating some minimal reprexes using `rd2html()`?

**Comment by jayhesselberth** [comment]
Here are some reprexes. `rd2html` messes with newlines, so the lower level view may be helpful. Not sure the last example generates a warning.

``` r
library(pkgdown)
library(magrittr)

# rd2html strips newlines, use low-level funcs too
rd_text <- pkgdown:::rd_text
as_html <- pkgdown:::as_html

# example 1 --------------------------------------

x <- "
Hello

testing

\\strong{para 1

para 2}"

rd2html(x)
#> [1] "Hello"           ""                "testing"         ""               
#> [5] "<strong>para 1"  ""                "para 2</strong>"
rd_text(x) %>% as_html()
#> [1] "\nHello\n\ntesting\n\n<strong>para 1\n\npara 2</strong>\n"

# example 2 --------------------------------------

x <- "\\preformatted{
line 1

line 2

line 3
}"

rd2html(x)
#> [1] "<pre>"  "line 1" ""       "line 2" ""       "line 3" "</pre>"
rd_text(x) %>% as_html()
#> [1] "<pre>\nline 1\n\nline 2\n\nline 3\n</pre>\n"

# example 3 --------------------------------------

x <- "\\section{A section}{
\\subsection{A subsection}{
  First para
  
  Second para
}
}"

rd2html(x)
#> Warning in tools::parse_Rd(con, fragment = fragment, encoding = "UTF-8"):
#> <connection>:1: unexpected section header '\section'
#> [1] "A section"             "<h3>A subsection</h3>" ""                     
#> [4] "First para"            ""                      "Second para"
rd_text(x) %>% as_html()
#> Warning in tools::parse_Rd(con, fragment = fragment, encoding = "UTF-8"):
#> <connection>:1: unexpected section header '\section'
#> [1] "A section\n<h3>A subsection</h3>\n\n  First para\n  \n  Second para\n\n\n"
```

<sup>Created on 2019-05-18 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by gustavdelius** [comment]
You can also see this in the documentation of pkgdown itself. Take a look at how [this](https://github.com/r-lib/pkgdown/blob/081639735104a03c01527f568a99f0ef7351433d/R/build.r#L125-L135) is rendered [here](https://pkgdown.r-lib.org/reference/build_site.html#yaml-config-template)

**Comment by jayhesselberth** [comment]
Rendering of verbatim blocks was fixed in #1076, but this doesn't fix subsection rendering.

**Comment by jayhesselberth** [comment]
`tag_subsection` blocks need to be handled specially in `flatten_para`.

The current HTML output looks like:

```html
<h3>A subsection</h3>
 
First paragraph

Second paragraph
```
but needs to look like:
```html
<h3>A subsection</h3>
<p>
First paragraph
</p>

<p>
Second paragraph
</p>
```
